packages/yarnpkg-core/sources/scriptUtils.ts

```diff
@@ -287,7 +291,7 @@ export async function prepareExternalProject(cwd: PortablePath, outputPath: Port
             // Run an install; we can't avoid it unless we inspect the
             // package.json, which I don't want to do to keep the codebase
             // clean (even if it has a slight perf cost when cloning v1 repos)
-            const install = await execUtils.pipevp(`yarn`, [`install`], {cwd, env, stdin, stdout, stderr, end: execUtils.EndStrategy.ErrorCode});
+            const install = await execUtils.pipevp(`yarn`, [`install`, `--mutex`, mutexPath], {cwd, env, stdin, stdout, stderr, end: execUtils.EndStrategy.ErrorCode});
```

Comment:
Could we instead limit it to one classic install at a time?

Reply:
Line 216 has a limit
```
const MAX_PREPARE_CONCURRENCY = 2;
const prepareLimit = pLimit(MAX_PREPARE_CONCURRENCY);
```
which could just be set to `1` to avoid this problem, but it isn't separated by package manager type, so setting it to 1 would have a negative perf impact on all installs that use git repos. I wasn't sure the tradeoff was worth it.

Reply:
We can add another limit for just Yarn classic installs.

Reply:
That would be a nice way to handle this concurrency issue without having to pass the mutex. I'd certainly prefer to handle the limit in yarn modern.
It seemed like it might be awkward to structure that file to have different limits, since the whole thing is wrapped in one `pLimit`.
I wasn't sure how make the `pLimit` more conditional like that... code suggestions welcome.